### PR TITLE
dwl: 0.5 -> 0.6

### PR DESCRIPTION
**Pull request by AndersonTorres**
## Description of changes

Closes https://github.com/NixOS/nixpkgs/pull/330317

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/dw/dwl/package.nix [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+  libxcb,
+  libxkbcommon,
+  pixman,
+  pkg-config,
+  stdenv,
+  testers,
+  wayland,
+  wayland-protocols,
+  wayland-scanner,
+  wlroots,
```
Comment (Rutherther) [inline pkgs/by-name/dw/dwl/package.nix:16]: dwl 0.6 depends on wlroots 0.17, the default one will be bumped to 0.18 by #330630, so dwl won't build after both branches are merged.

File: pkgs/by-name/dw/dwl/package.nix [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+    description = "Dynamic window manager for Wayland";
+    longDescription = ''
+      dwl is a compact, hackable compositor for Wayland based on wlroots. It is
+      intended to fill the same space in the Wayland world that dwm does in X11,
+      primarily in terms of philosophy, and secondarily in terms of
+      functionality. Like dwm, dwl is:
+
+      - Easy to understand, hack on, and extend with patches
+      - One C source file (or a very small number) configurable via config.h
+      - Limited to 2000 SLOC to promote hackability
```
Comment (Rutherther) [inline pkgs/by-name/dw/dwl/package.nix:116]: I am not sure the description is super important. But maybe it would be good to update it. This is not true anymore. From the readme:
> Given the Wayland architecture, dwl has to implement features from dwm and the xorg-server. Because of this, it is impossible to maintain the original project goal of 2000 SLOC and have a reasonably complete compositor with features comparable to dwm. However, this does not mean that the code will grow indiscriminately. We will try to keep the code as small as possible.

File: pkgs/by-name/dw/dwl/package.nix [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+
+  postPatch =
+    let
+      configFile =
+        if lib.isDerivation configDefH || builtins.isPath configDefH then
+          configDefH
+        else
+          writeText "config.def.h" configDefH;
+    in
+    lib.optionalString withCustomConfigDefH "cp ${configFile} config.def.h";
```
Comment (Rutherther) [inline pkgs/by-name/dw/dwl/package.nix:78]: What is the reason for this change away from conf argument? Wouldn't it make more sense to keep only conf, but copy it to `config.h` instead of `config.def.h`? To really override the config every time, not only if the source doesn't have config.h.

File: pkgs/by-name/dw/dwl/package.nix (hunk 1 above)
Comment (AndersonTorres) [reply pkgs/by-name/dw/dwl/package.nix:16]: The default is set at all-packages.nix.

File: pkgs/by-name/dw/dwl/package.nix (hunk 2 above)
Comment (AndersonTorres) [reply pkgs/by-name/dw/dwl/package.nix:116]: Noted.
Fixed.

File: pkgs/by-name/dw/dwl/package.nix (hunk 3 above)
Comment (AndersonTorres) [reply pkgs/by-name/dw/dwl/package.nix]: I am not understanding you here.
I have changed the `conf` argument to the more descriptive `configDefH` (now configH).
Your suggestion is then to move this configuration directly at `config.h`, instead of relying on the upstream build scripts that will move `config.def.h`? Well, it looks OK to me.

Comment (Rutherther) [reply pkgs/by-name/dw/dwl/package.nix]: It's not about relying on those scripts. It's about this: imagine you have a `config.h` in your repository. Then you fetch it by overriding `src` of dwl in nixpkgs, and set `configH`, expecting you will get the config you set in `configH`. But you won't, because it goes to `config.def.h` that is copied only if `config.h` doesn't exist.

I don't mind much if one or the other way is used. On the other hand it looks very good to me to either choose `configDefH` (or `configDefaultH` to be more clear) to override `config.def.h`, or `configH` to override `config.h`. How it is right now, called `configH`, but setting `config.def.h` is confusing to me, the same as `conf` setting `config.def.h` was confusing to me as well.

Comment (AndersonTorres) [reply pkgs/by-name/dw/dwl/package.nix]: I never thought this way. I will do this.

File: pkgs/by-name/dw/dwl/package.nix [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+  enableXWayland ? true,
+  withCustomConfigH ? (configH != null || conf != null),
+  # Configurable options
+  configH ?
+    if conf != null then lib.warn "conf parameter is deprecated; use configH instead" conf else null,
+  conf ? null,
+}:
+
+# If we set withCustomConfigH, let's not forget configH
+assert withCustomConfigH -> (configH != null || conf != null);
```
Comment (Rutherther) [inline pkgs/by-name/dw/dwl/package.nix:30]: I am thinking only `configH` should be used, since if someone decided to pass `configH = null; conf = {file}`, there would be `null` used instead of config file, but this check would pass. Maybe even asserting that if `configH` and `conf` are both set, they have to be the same, would make sense here. (or at least a warning) To prevent confusion with preference of those arguments.

**Review by Rutherther — CHANGES_REQUESTED** [review]

File: pkgs/by-name/dw/dwl/package.nix (hunk 4 above)
Comment (AndersonTorres) [reply pkgs/by-name/dw/dwl/package.nix]: Done.

**Review by Rutherther — APPROVED** [review]

**Review by wineee — APPROVED** [review]
